cdap-common/src/test/java/io/cdap/cdap/common/metadata/QueryParserTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.metadata;
+
+import io.cdap.cdap.common.metadata.QueryTerm.Qualifier;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.ArrayList;
+import java.util.List;
+
+public class QueryParserTest {
```

Comment:
to be complete, you should have more test cases and cover edge cases

- single term
- +*
- other white space than blanks
- empty query

etc.
One way to do that could be to define an array or a map of all test cases string -> expected result. And then have one test method that iterates over all of these entries and tests them.

Reply:
More test cases have been added to cover these.
Regarding the "+*" case, since the interpretation of a wildcard is left to the individual users of QueryParser, would it be better to handle that case within the users' tests?